firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java

```diff
@@ -315,13 +304,37 @@ private URL getUrl() {
 
   /** Create HTTP connection and set headers. */
   @SuppressLint("VisibleForTests")
-  public HttpURLConnection createRealtimeConnection() throws IOException {
-    URL realtimeUrl = getUrl();
-    HttpURLConnection httpURLConnection = (HttpURLConnection) realtimeUrl.openConnection();
-    setCommonRequestHeaders(httpURLConnection);
-    setRequestParams(httpURLConnection);
-
-    return httpURLConnection;
+  public Task<HttpURLConnection> createRealtimeConnection() {
+    // Make async call to get auth token.
+    Task<InstallationTokenResult> installationAuthTokenTask = firebaseInstallations.getToken(false);
+    // When the auth token task has finished, set up HTTP connection with headers and params.
+    return Tasks.whenAllComplete(installationAuthTokenTask)
+        .continueWithTask(
+            this.scheduledExecutorService,
+            (completedInstallationTask) -> {
+              if (!installationAuthTokenTask.isSuccessful()) {
+                return Tasks.forException(
+                    new FirebaseRemoteConfigClientException(
+                        "Firebase Installations failed to get installation ID for real-time.",
```

Comment:
I think the exception should be `installation auth token` here

Reply:
updated